### ISSUE TITLE
Error: Table 'glpidb.glpi_plugin_datainjection_entityinjections' doesn't exist

### ISSUE THREAD

**Issue by fforregs** [pluginsGLPI/datainjection#149]
GLPI 9.4, trying to import entites.
Settings:
![image](https://user-images.githubusercontent.com/9361966/53806127-32e6b680-3f5d-11e9-96d2-1156a6635ad3.png)
![image](https://user-images.githubusercontent.com/9361966/53806158-472ab380-3f5d-11e9-8cc4-e049fcfcb239.png)

sample data file: 
[new10.txt](https://github.com/pluginsGLPI/datainjection/files/2930925/new10.txt)

errors in log:
`[2019-03-05 12:27:31] glpisqllog.ERROR: DBmysql::query() in /var/www/glpi/inc/dbmysql.class.php line 177
  *** MySQL query error:
  SQL: SELECT MAX(`id`)+1 AS newID FROM `glpi_plugin_datainjection_entityinjections`
  Error: Table 'glpidb.glpi_plugin_datainjection_entityinjections' doesn't exist
  Backtrace :
  inc/dbmysqliterator.class.php:95
  inc/dbmysql.class.php:569                          DBmysqlIterator->execute()
  inc/entity.class.php:256                           DBmysql->request()
  inc/commondbtm.class.php:1100                      Entity->prepareInputForAdd()
  inc/commondropdown.class.php:752                   CommonDBTM->add()
  ...datainjection/inc/entityinjection.class.php:137 CommonDropdown->import()
  ...injection/inc/commoninjectionlib.class.php:1524 PluginDatainjectionEntityInjection->customimport()
  ...injection/inc/commoninjectionlib.class.php:1446 PluginDatainjectionCommonInjectionLib->effectiveAddOrUpdate()
  .../datainjection/inc/entityinjection.class.php:85 PluginDatainjectionCommonInjectionLib->processAddOrUpdate()
  plugins/datainjection/inc/engine.class.php:147     PluginDatainjectionEntityInjection->addOrUpdateObject()
  ...datainjection/inc/clientinjection.class.php:267 PluginDatainjectionEngine->injectLine()
  ...datainjection/inc/clientinjection.class.php:219 PluginDatainjectionClientInjection::processInjection()
  ...datainjection/front/clientinjection.form.php:40 PluginDatainjectionClientInjection::showInjectionForm()
  {"user":"2@glpi","mem_usage":"0.004\", 3.04Mio)"}
[2019-03-05 12:27:31] glpisqllog.ERROR: DBmysql::query() in /var/www/glpi/inc/dbmysql.class.php line 177
  *** MySQL query error:
  SQL: INSERT INTO `glpi_entities` (`country`, `email`, `level`, `name`, `completename`, `entities_id`, `max_closedate`, `date_creation`, `date_mod`) 
                            VALUES ('rus', 'mysoft2@localhost', '2', 'soft2', 'Root entity > soft2', '0', '2019-03-05 12:27:31', '2019-03-05 12:27:31', '2019-03-05 12:27:31')
  Error: Duplicate entry '0' for key 'PRIMARY'
  Backtrace :
  inc/dbmysql.class.php:821
  inc/commondbtm.class.php:658                       DBmysql->insert()
  inc/commondbtm.class.php:1140                      CommonDBTM->addToDB()
  inc/commondropdown.class.php:752                   CommonDBTM->add()
  ...datainjection/inc/entityinjection.class.php:137 CommonDropdown->import()
  ...injection/inc/commoninjectionlib.class.php:1524 PluginDatainjectionEntityInjection->customimport()
  ...injection/inc/commoninjectionlib.class.php:1446 PluginDatainjectionCommonInjectionLib->effectiveAddOrUpdate()
  .../datainjection/inc/entityinjection.class.php:85 PluginDatainjectionCommonInjectionLib->processAddOrUpdate()
  plugins/datainjection/inc/engine.class.php:147     PluginDatainjectionEntityInjection->addOrUpdateObject()
  ...datainjection/inc/clientinjection.class.php:267 PluginDatainjectionEngine->injectLine()
  ...datainjection/inc/clientinjection.class.php:219 PluginDatainjectionClientInjection::processInjection()
  ...datainjection/front/clientinjection.form.php:40 PluginDatainjectionClientInjection::showInjectionForm()
  {"user":"2@glpi","mem_usage":"0.010\", 3.04Mio)"}`

**Comment by tuxxic** [comment]
Hi, I have the same problem. Is the generated SQL query wrong ?

`[2019-03-08 17:38:20] glpisqllog.ERROR: DBmysql::query() in /var/www/glpi/inc/dbmysql.class.php line 177
  *** MySQL query error:
  SQL: INSERT INTO `glpi_entities` (`level`, `name`, `completename`, `entities_id`, `max_closedate`, `date_creation`, `date_mod`) VALUES ('2', 'Test2', 'Root entity > Test2', '0', '2019-03-08 17:38:20', '2019-03-08 17:38:20', '2019-03-08 17:38:20')
  Error: Duplicate entry '0' for key 'PRIMARY'
  Backtrace :
  inc/dbmysql.class.php:821
  inc/commondbtm.class.php:658                       DBmysql->insert()
  inc/commondbtm.class.php:1140                      CommonDBTM->addToDB()
  inc/commondropdown.class.php:752                   CommonDBTM->add()
  ...datainjection/inc/entityinjection.class.php:137 CommonDropdown->import()
  ...injection/inc/commoninjectionlib.class.php:1524 PluginDatainjectionEntityInjection->customimport()
  ...injection/inc/commoninjectionlib.class.php:1446 PluginDatainjectionCommonInjectionLib->effectiveAddOrUpdate()
  .../datainjection/inc/entityinjection.class.php:85 PluginDatainjectionCommonInjectionLib->processAddOrUpdate()
  plugins/datainjection/inc/engine.class.php:147     PluginDatainjectionEntityInjection->addOrUpdateObject()
  ...datainjection/inc/clientinjection.class.php:267 PluginDatainjectionEngine->injectLine()
  ...datainjection/inc/clientinjection.class.php:219 PluginDatainjectionClientInjection::processInjection()
  ...datainjection/front/clientinjection.form.php:40 PluginDatainjectionClientInjection::showInjectionForm()
  {"user":"2@glpi","mem_usage":"0.007\", 2.72Mio)"}
`

**Comment by tuxxic** [comment]
@arhipovdv  do you have found what's wrong ?

**Comment by arhipovdv** [comment]
@tuxxic no :( data injections doesn't work only with Entities. Create them manualy. Creating Users (for example) with data injections works normal.

**Comment by tuxxic** [comment]
@arhipovdv thanks for the feedback ! I did a manual SQL to create my entities, since time was lacking to my project. But if you tell me that user import is going to go okay then, I will go on with the plugin

**Comment by felag** [comment]
Hello, I had the same problem and made an ugly temporary patch in GLPi core php code: in file **[…]/glpi/inc/entity.class.php** changed line 256:
```
         'FROM'   => self::getTable()
```

to
```
         'FROM'   => Entity::getTable()
```

then I made the import with web interface and put back original line 256 in place.


(I will not be around Internet connection for a couple of weeks, but I hope this helps to find a more valuable patch in plugin for future releases.)

**Comment by stonebuzz** [comment]
duplicate with https://github.com/pluginsGLPI/datainjection/issues/172

**Comment by stonebuzz** [comment]
I confirmed bug, @felag solution seems good

try this PR -> https://github.com/glpi-project/glpi/pull/6857

is for GLPI not datainjection

**Comment by cedric-anne** [comment]
Hi,

Fixed in GLPI 9.5.

Regards
